src/linechart/DataSeries.jsx

```diff
@@ -12,129 +11,90 @@ module.exports = React.createClass({
         .interpolate(props.interpolationType);
 
-    // Check whether or not an arbitrary data element
-    // is a date object (at index 0 here)
-    // If it's a date, then we set the x scale a bit differently
-    // NOTE: a sparse array could still break this, even with the length check above
-    if (this._isDate(props.data[0], xAccessor)) {
-      interpolatePath.x(function(d) {
-        return props.xScale(props.xAccessor(d).getTime());
-      });
-    } else {
-      interpolatePath.x(function(d) {
-        return props.xScale(props.xAccessor(d));
-      });
-    }
-
-    // Create an immstruct reference for the series name
-    // and set it to 'inactive'
-    props.structure.cursor('voronoiSeries').set(props.seriesName, 'inactive');
-
-    // Having set the Voronoi line series name cursor to 'inactive'
-    // We now pass on the Voronoi line series name reference to the
-    // *both* the line and circle component
-    var voronoiSeriesRef = props.structure.reference(['voronoiSeries', props.seriesName]);
-
-
-    var circles = null;
-
-    if (props.displayDataPoints) {
-      // Map over data to generate SVG circles at data points
-      // if datum is a date object, treat it a bit differently
-      circles = props.data.map(function(point, idx) {
-        var cx, cy;
-        if (this._isDate(point, xAccessor)) {
-          cx = props.xScale(xAccessor(point).getTime());
-        } else {
-          cx = props.xScale(xAccessor(point));
-        }
-        if (this._isDate(point, yAccessor)) {
-          cy = props.yScale(yAccessor(point).getTime());
+        if (this._isDate(props.data[0].values[0], xAccessor)) {
+          interpolatePath.x(function(d) {
+            return props.xScale(props.xAccessor(d).getTime());
+          });
         } else {
-          cy = props.yScale(yAccessor(point));
+          interpolatePath.x(function(d) {
+            return props.xScale(props.xAccessor(d));
+          });
         }
 
-        var id = props.seriesName + '-' + idx;
-
-        // Create an immstruct reference for the circle id
-        // and set it to 'inactive'
-        props.structure.cursor('voronoi').set(id, 'inactive');
-
-        // Having set the Voronoi circle id cursor to 'inactive'
-        // We now pass on the Voronoi circle id reference to the
-        // circle component, where it will be observed and dereferenced
-        var voronoiRef = props.structure.reference(['voronoi', id]);
-
-        return (
-          <Circle
-            voronoiRef={voronoiRef}
-            voronoiSeriesRef={voronoiSeriesRef}
-            structure={props.structure}
-            cx={cx}
-            cy={cy}
-            r={props.circleRadius}
-            fill={props.fill}
-            key={idx}
-            id={props.seriesName + '-' + idx}
+    var lines = props.data.map((series, idx) => {
+      return <Line 
+        path={interpolatePath(series.values)}
+        stroke={props.colors(props.colorAccessor(series.name, idx))}
```

Comment:
`series.name` here is incorrect. It should take the entire series (`series`). That way users can pass additional information into the series and use it to create colors, or do whatever else they want.